mongoengine/queryset/queryset.py

```diff
@@ -518,7 +518,9 @@ def update(self, upsert=False, multi=True, write_concern=None, **update):
             else:
                 update["$set"] = {"_cls": queryset._document._class_name}
         try:
-            with set_write_concern(queryset._collection, write_concern) as collection:
+            with set_read_write_concern(
+                queryset._collection, write_concern, read_concern
+            ) as collection:
```

Comment:
Note that this indeed will work well because we're using the return value of `collection` everywhere within this `with` block. This would be equivalent to writing this though:
```
collection = next(set_read_write_concern(queryset._collection, write_concern, read_concern))
update_func = collection.update_one
...
```